### PR TITLE
Remove lenient protocol

### PR DESCRIPTION
**Pull request by benrudolph**
@czue @wpride removes the ability to use `as=<user>@<domain>`. Requires full username. tested old cloudcare, new cloudcare, and edit forms. i think this simplifies many things

cc: @emord

### CODE REVIEW

File: corehq/apps/ota/utils.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from corehq.apps.users.util import format_username
 from dimagi.utils.logging import notify_exception
 
-from corehq.apps.users.models import CommCareUser, WebUser
+from corehq.apps.users.models import CommCareUser, WebUser, CouchUser
```
Comment (dimagimon) [inline corehq/apps/ota/utils.py]: F401 'CommCareUser' imported but unused
F401 'WebUser' imported but unused

**Review by czue — APPROVED** [review]
nice this is way simpler

**Comment by czue** [comment]
for some reason feeling a little gunshy about merging (just feels like the type of thing that might have unexpected side effects and we don't have great integration tests). if you're confident in your process I think go for it though.

**Comment by benrudolph** [comment]
also a bit tentative myself. this has already bitten me loads of times, each time though i've added a test case that had bitten me. i'll do another round of QA on staging and then report back

**Review by emord — APPROVED** [review]

**Comment by benrudolph** [comment]
Ok tested the possible end points that i know of locally and on staging and haven't seen any errors. pulling the trigger. will monitor it after deploy

**Pull request merged**